app/Platform/Enums/PlatformExecutionEnvironment.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Enums;
+
+enum PlatformExecutionEnvironment: string
+{
+    /** Windows, macOS, Linux */
+    case Desktop = 'desktop';
+
+    /** Android, iOS */
+    case Mobile = 'mobile';
```

Comment:
Would the SteamDeck be desktop or mobile? Does it matter? I think it qualifies as Linux.

Reply:
I also think we'd treat it as Linux.

Reply:
I'm thinking `PlatformExecutionEnvironment` mostly exists to help inform UI concerns, such as what icon gets displayed here:
<img width="178" alt="Screenshot 2025-03-24 at 4 11 27 PM" src="https://github.com/user-attachments/assets/577f4de3-3407-4e15-80d4-a00008ac9ea9" />

It's also nullable, as I suspect there may be cases where some platform may not fit neatly into one of these boxes.

For Steam Deck specifically, it's probably classed best as a desktop device, as despite its form factor, it is running a full desktop operating system.